### DISCUSSION TITLE
What metadata is being embeded when the data is send in chunks?

### DISCUSSION THREAD

**Discussion by Zenodeon** [jchristn/SuperSimpleTcp#187] [General]
I would use wattson tcp but it uses system.text.json so i am having a hard time making it work on android, (does not give any errors but just does not work but works on pc).

I saw your comment on one of the closed issues mentioning you send them with metadata to be framed by myself. I am petty new to this and would be helpful if you can provide the metadata structure and basic instructions on how to decode it and any helpful tips.

I also saw you recommended EasyTCP too in one of the closed issues which i will try after trying framing with this framework as i know it works on Android.

**Comment by jchristn** [comment]
Have you tried implementing your own ```ISerializationHelper``` and attaching it to the ```SerializationHelper``` property on both WatsonTcp client and server?

The framing used in Watson is described here: https://github.com/jchristn/WatsonTcp/blob/master/FRAMING.md

I wouldn't recommend building a state machine using SuperSimpleTcp.  It reads as much data as it can from the underlying socket (up to the size of the defined stream buffer) and doesn't have an understanding of logical message boundaries.  I'd use CavemanTcp or something else, that way you know you have explicit control over the amount of data being read.

**Comment by Zenodeon** [comment]
No, I did not try implementing ```ISerializationHelper ``` as I did not know it was possible, was using your ```logger``` property in SuperSimpleTCP and did not take extra time refracting that while moving to Wattson, but will keep this in mind for future usage.

I was looking for SuperSimpleTCP receiving Data Metadata but that link is useful too, will use it if I go with CavemanTCP.

Thanks for the recommendation and the backend explanation. I will look into EasyTCP Framework as it's already layout and try out CavemanTCP as last resort.

also, brilliant names for these frameworks 😂.

**Comment by Zenodeon** [comment]
Hey, I decided to go with EasyTCP as it's petty much what I needed and works well on Android as well, but had to implement my own system to keep it connected to the server as KeepServerAlive features is not supported in my usecase. 

Thank you for the recommendation and your quick help.

**Reply by jchristn** [reply]
Awesome!  I'm glad you found something that will work for you.  The owner/maintainer on that package is pretty spectacular, too.
